piet-cairo/src/lib.rs

```diff
@@ -524,4 +532,359 @@ impl TextLayout for CairoTextLayout {
     fn width(&self) -> f64 {
         self.font.text_extents(&self.text).x_advance
     }
+
+    // first assume one line.
+    // TODO do with lines
+    /// Hit Test for Point
```

Comment:
The authoritative docstring should be on the trait (in piet crate). If there are impl-specific things to add, they can go here, but I think discoverability will be lower. (this comment applies to most of the new trait impls)

Reply:
Improved docs